src/lib.rs

```diff
@@ -5585,6 +5594,17 @@ mod tests {
         assert_eq!(pipe.handshake(), Err(Error::UnknownVersion));
     }
 
+    #[test]
+    fn config_version_reserved() {
+        Config::new(0xbabababa).unwrap();
+        Config::new(0x00000000).unwrap();
+    }
+
+    #[test]
+    fn config_version_invalid() {
+        assert_eq!(Config::new(0xb1bababa).err().unwrap(), Error::UnknownVersion);
```

Comment:
used `err().unwrap()` as the `Config` doesn't declare `#[derive(Debug)]`. Not sure if there is a reason for this. If not we could do so and use `unwrap_err()`